tests/integration/states/test_file.py

```diff
@@ -1273,11 +1273,14 @@ def test_recurse_issue_40578(self, name):
                              name=name,
                              source='salt://соль')
         self.assertSaltTrueReturn(ret)
+        # Providing unicode to os.listdir so that we avoid having listdir try
+        # to decode the filenames using the systemencoding on windows.
         self.assertEqual(
-            sorted(salt.utils.data.decode(os.listdir(name), normalize=True)),
+            sorted(os.listdir(name.decode('utf-8'))),
```

Comment:
For py2 windows, If we pass bytes to os.listdir it will use the system encoding to try and return bytes. I think we just need to make this a special case for PY2 windows for now.

Reply:
OK cool, I agree then. I'll re-review once you've had a chance to make the change, other than this one thing everything else looked good to merge.